docs/docs/api/http.md

```diff
@@ -352,13 +352,42 @@ This is a [paginated](#pagination) endpoint. Messages are sorted from oldest to
 
 Sends a message to a conversation and returns a payload.
 
-**Sample request**
+**Sending a text message**
+
+```json5
+{
+  "conversation_id": "a688d36c-a85e-44af-bc02-4248c2c97622",
+  "message": {
+    "text": "Hello World",
+    "type": "text" 
+  }
+}
+```
+
+**Sending an attachment message**
+
+```json5
+{
+  "conversation_id": "a688d36c-a85e-44af-bc02-4248c2c97622",
+  "message": {
+    "url": "http://example.org/myfile",
+    "type": "image|video|audio|file" 
+  }
+}
+```
+
+**Sending source templates**
+
+Some sources support sending templates, which can be used to display rich content such as buttons or cards. You
+can send source templates by setting the type to `source.template`. Please refer to the source documentation
+to see the expected values for the `payload` field.
 
 ```json5
 {
   "conversation_id": "a688d36c-a85e-44af-bc02-4248c2c97622",
   "message": {
-    "text": "{String}"
+    "payload": "{\"template_type\":\"buttons\",\"buttons\":[\"Welcome to our shop\"]}",
+    "type": "source.template" 
```

Comment:
@airyhq/codeowners-frontend The idea is that the backend does not maintain the data model of the sources, because it changes frequently and requires a lot of code that would in part be duplicated by the rendering on the frontend.

**Example**
To make this more concrete here is how this would work for Facebook (TODO #644):
```json
{
  "conversation_id": "a688d36c-a85e-44af-bc02-4248c2c97622",
  "message": {
    "type": "source.template",
    "payload": {
      "template_type":"generic",
      "elements":[
        {
          "title":"<TITLE_TEXT>",
          "image_url":"<IMAGE_URL_TO_DISPLAY>",
          "subtitle":"<SUBTITLE_TEXT>",
          "default_action": {
            "type": "web_url",
            "url": "<DEFAULT_URL_TO_OPEN>",
            "webview_height_ratio": "<COMPACT | TALL | FULL>"
          },
          "buttons": []
        }
      ]
    }
  }
}
```
Here the `payload` field is the same as the `payload` [you provide](https://developers.facebook.com/docs/messenger-platform/send-messages/template/generic/) when sending a template to Facebook directly.

As you can already imagine, the exact use of the `payload` field is dependent on each source and will be documented in the source docs. The implementation and documentation of the above payload will happen in #644

**Getting templates from the API**
The backend will store this as a stringified version and return it on messaging apis (`messsage.list`, websocket etc.) like so:
```
{
  ...
  "content": [
    {
      "type": "source.template",
      "payload": "{\"template_type\":\"generic\",\"elements\":[{\"title\":\"<TITLE_TEXT>\",\"image_url\":\"<IMAGE_URL_TO_DISPLAY>\",\"subtitle\":\"<SUBTITLE_TEXT>\",\"default_action\":{\"type\":\"web_url\",\"url\":\"<DEFAULT_URL_TO_OPEN>\",\"webview_height_ratio\":\"<COMPACT | TALL | FULL>\"},\"buttons\":[]}]}"
    }
  ]
}
```

Rendering one of those templates will be implemented in #643

**Google example**

```json
{
  "conversation_id": "...",
  "message": {
    "type": "source.template",
    "payload": {
      "suggestions": [
        {
          "reply": {
            "text": "Main Menu",
            "postbackData": "main-menu"
          }
        },
        {
          "action": {
            "text": "Some Link",
            "postbackData": "tapped-link1",
            "openUrlAction": {
              "url": "https://www.google.com"
            }
          }
        }
      ]
    }
  }
}
```

Reply:
My question is: If the structure of the sources changes a lot, wouldn't this also fail when they change it? We will have to change the code every time they change something right?

Reply:
@AitorAlgorta that's true and the reason why we would like to only handle this code on the frontend, where it needs to be rendered.

Reply:
By the way, this is what the `SourceTemplate` typescript interface would look like:
```

export interface SourceTemplate extends Content {
    type: "source.template";
    payload: any;
}
```

Reply:
The problem for me: if we don't clean it up in the backend ,we need to change all frontends for every change. Is it possible to have an api where the changes are abstracted fully in the backend? That way we can react to changes quicker and don't have to invest in coding 3 solutions and deploying all of them.